.pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml

```diff
@@ -166,6 +170,33 @@ steps:
     name: "validatePods"
     displayName: "Validate Pods"
 
+  - script: |
+      if [ "$CILIUM_VERSION_TAG" = "cilium-nightly-pipeline" ]; then 
+        kubectl get pod -owide -n cilium-test
+        echo "wait for pod and cilium identity deletion"
+        ns="cilium-test"
+        while true; do
+          pods=$(kubectl get pods -n $ns --no-headers=true 2>/dev/null)
+          if [[ -z "$pods" ]]; then
+            echo "No pods found"
+              break
+          fi
+          sleep 2s
+        done
+        sleep 20s
+        echo "Verify cilium identities are deleted"
+        checkIdentity="$(kubectl get ciliumidentity -o json | grep cilium-test | jq -e 'length == 0')"
+        if [[ -n $checkIdentity ]]; then
+          echo "##[error]Cilium Identities still present"
+        else
+          printf -- "Identities deleted\n"
```

Comment:
Can we have additional logging here? If identities are still present -> see what they are and confirm which identities are being deleted (if possible).

Reply:
Sure, the check is verifying that the identities are being deleted from the cilium-test ns. I updated the logging with references to the namespace.